task/build-image-manifest/0.1/build-image-manifest.yaml

```
@@ -105,3 +105,38 @@ spec:
       capabilities:
         add:
           - SETFCAP
+
+  - name: push-dockerfile
+    image: quay.io/konflux-ci/oras@sha256:5d0a8a5535fcc4ba467264cacbdeab2fb8662a538a61cb7fc8b3155e3f20fa39
+    env:
+    - name: IMAGE_DIGEST_FILE
+      value: $(results.IMAGE_DIGEST.path)
+    script: |
+      suffix=".dockerfile"
+      image_without_tag=${IMAGE%:*}
+      # remove the potential existing newline
+      manifest_digest=$(cat "${IMAGE_DIGEST_FILE}" | tr -d '\r\n')
+      dockerfile_image=${image_without_tag}:${manifest_digest/:/-}${suffix}
+      artifact_type=application/vnd.konflux.dockerfile
+
+      # Select a Dockerfile from one of the arch
+      arch_image_digest=$(oras manifest fetch "$IMAGE" | yq -P -oy | yq '.manifests[0].digest')
+      output_dir=$(mktemp -d)
+      retry oras pull --output "$output_dir" "${image_without_tag}:${arch_image_digest/:/-}${suffix}"
+
+      echo "Selecting auth for $IMAGE"
+      auth_json=$(mktemp)
+      select-oci-auth $IMAGE >"$auth_json"
+
+      # Do not assume the actual file name. Generally, it could be Dockerfile or Containerfile.
+      cd "$output_dir"
+      n=1
+      find . -type f | while read filename; do
```

```suggestion
      find . -type f | while read -r filename; do
```
otherwise `read` treats some characters specially